FreeRTOS_Sockets.c

```diff
@@ -1599,23 +1599,44 @@ void * vSocketClose( FreeRTOS_Socket_t * pxSocket )
         FreeRTOS_Socket_t * pxOtherSocket;
         uint16_t usLocalPort = pxSocketToDelete->usLocalPort;
 
-        for( pxIterator = listGET_NEXT( pxEnd );
-             pxIterator != pxEnd;
-             pxIterator = listGET_NEXT( pxIterator ) )
+        if( pxSocketToDelete->u.xTCP.ucTCPState == ( uint8_t ) eTCP_LISTEN )
         {
-            pxOtherSocket = ipCAST_PTR_TO_TYPE_PTR( FreeRTOS_Socket_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
+            for( pxIterator = listGET_NEXT( pxEnd );
+                 pxIterator != pxEnd; )
+            {
+                pxOtherSocket = ipCAST_PTR_TO_TYPE_PTR( FreeRTOS_Socket_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
 
-            if( ( pxOtherSocket->u.xTCP.ucTCPState == ( uint8_t ) eTCP_LISTEN ) &&
-                ( pxOtherSocket->usLocalPort == usLocalPort ) &&
-                ( pxOtherSocket->u.xTCP.usChildCount != 0U ) )
+                /* This needs to be done here, before calling vSocketClose. */
+                pxIterator = listGET_NEXT( pxIterator );
+
+                if( ( pxOtherSocket->u.xTCP.ucTCPState != ( uint8_t ) eTCP_LISTEN ) &&
+                    ( pxOtherSocket->usLocalPort == usLocalPort ) &&
+                    ( pxOtherSocket->u.xTCP.bits.bPassQueued || pxOtherSocket->u.xTCP.bits.bPassAccept ) )
```

Comment:
For me the expression is perfect, but MISRA would like to see:
~~~c
( ( pxOtherSocket->u.xTCP.bits.bPassQueued != 0U ) ||
  ( pxOtherSocket->u.xTCP.bits.bPassAccept != 0U ) )
~~~

Reply:
okay, done.